### PR TITLE
[MID-3] add header footer components

### PR DESCRIPTION
**Pull request by Yivgha**
+tailwind styles;
+reusable btn component;

trello: https://trello.com/c/Ll5qjjEM

env: https://mayak-2-h8fflutym-yivghas-projects.vercel.app/

### CODE REVIEW

File: src/app/_components/Footer/Footer.js [hunk 1]
```diff
@@ -1,5 +1,101 @@
...
+  const basicLink = "no-underline list-none cursor-pointer";
+  const iconSizes = "w-[24px] h-[24px]";
+  const iconColors = "text-other-white hover:text-primary-400";
+  const transition = "transition duration-200 ease-in-out";
+  const footerBtnHoverFocus =
+    "hover:bg-gray-100 hover:text-primary-500 focus:bg-gray-100 focus:text-primary-600 focus:border-gray-100 active:bg-other-white active:text-primary-600 active:border-other-white";
+
+  return (
+    <footer className="text-white flex w-full flex-col bg-primary-800 px-20 py-12">
+      <div className={`${flexBetween}`}>
```
Comment (Dimirolz) [inline src/app/_components/Footer/Footer.js]: @Yivgha let's refactor all the class interpolation into the usage of the `cn` utility.

Take inspiration from here -> https://github.com/shadcn-ui/ui/blob/0fae3fd93ae749aca708bdfbbbeddc5d576bfb2e/templates/next-template/lib/utils.ts#L2

TwMerge is an important piece of the workflow with Tailwind. It fixes all the problems related to Tailwind style conflicts.

File: src/app/_components/Footer/Footer.js [hunk 2]
```diff
@@ -1,5 +1,101 @@
...
+      </div>
+      <div className="my-12 h-[1px] w-full bg-gray-300"></div>
+      <div className={`${flexBetween}`}>
+        <p className="text-p4 font-normal text-other-white">Маяк @ 2024</p>
+        <div className={`${flexCenter}`}>
+          <p className="mr-4 text-p2 font-medium text-other-white">
+            Слідкуй за нами в соцмережах
+          </p>
+
+          <Link
```
Comment (Dimirolz) [inline src/app/_components/Footer/Footer.js]: @Yivgha let's extract this link repetition into a loop. You can create an array, socialLinks, with objects that contain information for each link.

The `mr-4` between links is doing the job, but it's more convenient to use `gap-4` and put the links inside a flex container.

File: src/app/_components/Header/Header.js [hunk 3]
```diff
@@ -1,5 +1,110 @@
...
+            <Link
+              href="/events"
+              aria-label="Open events page on this link click"
+              className={`${basicLink}`}
+            >
+              <p className={`${listItemText} ${transition}`}>Події</p>
+            </Link>
+          </li>
+        </ul>
+        <div className={`${flexBetween} gap-6`}>
```
Comment (Dimirolz) [inline src/app/_components/Header/Header.js]: @Yivgha I saw similar links in the footer, so I suggest you extract this markup with social links into a separate component, like `SocialLinks`.

File: src/app/_components/Icons/Facebook.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import p from 'prop-types';
+
+const FacebookComponent = ({ className }) => (
```
Comment (Dimirolz) [inline src/app/_components/Icons/Facebook.js]: @Yivgha for such cases, you can often see the usage of `svgr `in various projects. I don't have a strong opinion there; `svgr `can be tricky to set up sometimes.

You can discuss the need for `svgr ` in such cases with your colleagues.

File: src/app/_components/OutlinedBtn/OutlinedBtn.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  const transition = 'transition duration-200 ease-in-out';
+  const basicBtn = 'w-[185px] h-[40px] py-2.5 px-6 rounded-full bg-transparent outline-none border cursor-pointer text-p4 font-bold text-center';
+  const btnHoverFocus = 'hover:bg-primary-200 focus:text-primary-600  focus:bg-primary-200 focus:border-primary-600 active:bg-primary-200 active:text-primary-600 active:border-gray-700 disabled:text-gray-400 disabled:bg-transparent disabled:border-gray-200';
+
+  return (
+    <button
+      type="button"
+      className={`${className} ${flexCenter} ${basicBtn} ${btnHoverFocus} ${transition}`}
+    >
+      {' '}
```
Comment (Dimirolz) [inline src/app/_components/OutlinedBtn/OutlinedBtn.js]: @Yivgha I'm not sure what the reason is for those white spaces. It would be better to remove them from the base component.

**Review by Dimirolz — CHANGES_REQUESTED** [review]
Please resolve the conflicts of this branch with the current `dev` branch

File: src/app/_components/Footer/Footer.js [hunk 6]
```diff
@@ -1,5 +1,101 @@
-import React from 'react';
+"use client";
```
Comment (alex-popov-tech) [inline src/app/_components/Footer/Footer.js]: 🔴 why is it client component? i don't see any client specific features used here

File: src/app/_components/Footer/Footer.js [hunk 7]
```diff
@@ -1,5 +1,101 @@
...
+
+import React from "react";
+import Image from "next/image";
+import Link from "next/link";
+import OutlinedBtn from "../OutlinedBtn";
+import * as FooterImg from "../../../../public/assets/icons/white-logo.svg";
+import InstagramComponent from "../Icons/Instagram";
+import FacebookComponent from "../Icons/Facebook";
+import XComponent from "../Icons/XSocial";
```
Comment (alex-popov-tech) [inline src/app/_components/Footer/Footer.js]: 🟠 can these imports be prettified with import aliases?

File: src/app/_components/Header/Header.js [hunk 8]
```diff
@@ -1,5 +1,110 @@
+'use client';
```
Comment (alex-popov-tech) [inline src/app/_components/Header/Header.js]: 🟠 comments about footer are relevant for header component too

File: src/app/_components/Icons/Facebook.js (hunk 4 above)
Comment (alex-popov-tech) [inline src/app/_components/Icons/Facebook.js]: 🟢 can't we use something like react-icons for this?

File: src/app/_components/OutlinedBtn/index.js [hunk 9]
```diff
@@ -0,0 +1 @@
+export { default } from './OutlinedBtn';
```
Comment (alex-popov-tech) [inline src/app/_components/OutlinedBtn/index.js]: 🟠 what is the point for component of 20 lines long to create folder and 2 files?

File: src/app/admin/adminApp.js [hunk 10]
```diff
@@ -38,13 +38,13 @@ export default function AdminPage() {
...
       <Resource
-        name='Therapy'
+        name="Therapy"
         list={ListGuesser}
         edit={EditGuesser}
         show={ShowGuesser}
       />
       <Resource
-        name='Specialist'
+        name="Specialist"
```
Comment (alex-popov-tech) [inline src/app/admin/adminApp.js]: 🟠 is that according to linter rools? next file uses single quotes, i suspect there is something going on :)

**Review by alex-popov-tech — CHANGES_REQUESTED** [review]

File: src/app/_components/Footer/Footer.js (hunk 1 above)
Comment (Yivgha) [reply src/app/_components/Footer/Footer.js]: Good advice! Created cn util fn.

File: src/app/_components/OutlinedBtn/OutlinedBtn.js (hunk 5 above)
Comment (Yivgha) [reply src/app/_components/OutlinedBtn/OutlinedBtn.js]: It was linter fix, deleted.

File: src/app/_components/Footer/Footer.js (hunk 6 above)
Comment (Yivgha) [reply src/app/_components/Footer/Footer.js]: Deleted.

File: src/app/_components/Footer/Footer.js (hunk 7 above)
Comment (Yivgha) [reply src/app/_components/Footer/Footer.js]: added some aliases

File: src/app/_components/Header/Header.js (hunk 8 above)
Comment (Yivgha) [reply src/app/_components/Header/Header.js]: Deleted.

File: src/app/_components/Icons/Facebook.js (hunk 4 above)
Comment (Yivgha) [reply src/app/_components/Icons/Facebook.js]: I'd like to, but react-icons don't have all icons we need.

File: src/app/_components/OutlinedBtn/index.js (hunk 9 above)
Comment (Yivgha) [reply src/app/_components/OutlinedBtn/index.js]: Deleted.

File: src/app/admin/adminApp.js (hunk 10 above)
Comment (Yivgha) [reply src/app/admin/adminApp.js]: added some .prettierrc.json changes

File: src/app/_components/Footer/Footer.js (hunk 2 above)
Comment (Yivgha) [reply src/app/_components/Footer/Footer.js]: created reusable link components

File: src/app/_components/Header/Header.js (hunk 3 above)
Comment (Yivgha) [reply src/app/_components/Header/Header.js]: done

File: src/app/_components/Icons/Facebook.js (hunk 4 above)
Comment (Yivgha) [reply src/app/_components/Icons/Facebook.js]: implemented svgr

File: jsconfig.json [hunk 11]
```diff
@@ -1,7 +1,13 @@
...
   "compilerOptions": {
     "paths": {
-      "@/*": ["./src/*"]
+      "@/*": ["./src/*"],
+      "@icons/*": ["./public/assets/icons/*"],
+      "@outlinedBtn/*": ["./src/app/_components/OutlinedBtn/*"],
+      "@utils/*": ["./src/app/utils/*"],
+      "@config/*": ["./src/app/config/*"],
+      "@socialLink/*": ["./src/app/_components/SocialLink/*"],
+      "@innerLink/*": ["./src/app/_components/InnerLink/*"]
```
Comment (Dimirolz) [inline jsconfig.json]: @Yivgha you can create links only to _components, not for each individual component, and have paths like `@components/SocialLink`. It's more convenient than creating them for each component.

File: src/app/_components/SocialLink/SocialLink.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+          if (a.title < b.title) {
+            return -1;
+          }
+          if (a.title > b.title) {
+            return 1;
+          }
+          return 0;
+        })
+        .map((link, idx) => (
+          <Link
```
Comment (Dimirolz) [inline src/app/_components/SocialLink/SocialLink.js]: @Yivgha Let's refactor this component in a few steps.

1. First, let's make it dumb. Simply render links without any sorting or branching logic. Move the sorting logic into the `Footer` component.
2. Remove `status` prop
3. Receive only the necessary props. In this case, accept only the links objects. Not whole nav object.
4. Replace the usage of `.sort` with `.toSorted` (or clone array before sorting, like [...arr].sort) since it doesn't create side effects by mutating the original array.

**Review by Dimirolz — CHANGES_REQUESTED** [review]

File: next.config.mjs [hunk 13]
```diff
@@ -11,6 +11,14 @@ const nextConfig = {
   eslint: {
     ignoreDuringBuilds: true,
   },
+  webpack(config) {
+    config.module.rules.push({
```
Comment (alex-popov-tech) [inline next.config.mjs:15]: 🟢 would be really nice to have comments describing purpose of such changes

File: src/app/_components/Footer/Footer.js [hunk 14]
```diff
@@ -1,9 +1,58 @@
...
 
 export default function Footer() {
+  // Basic styles
+  const flexBetween = 'inline-flex flex-row items-center justify-between';
+  const flexCenter = 'inline-flex flex-row items-center justify-center';
+  const basicLink = 'no-underline list-none cursor-pointer';
+  const iconColors = 'text-other-white hover:text-primary-400';
+  const transition = 'transition duration-200 ease-in-out';
+  const footerBtnHover = 'hover:bg-gray-100 hover:text-primary-500  ';
+  const footerBtnFocus = 'focus:bg-gray-100 focus:text-primary-600 ';
```
Comment (alex-popov-tech) [inline src/app/_components/Footer/Footer.js]: 🟢 there are trailing spaces in strings

File: src/app/_components/SocialLink/SocialLink.js (hunk 12 above)
Comment (Yivgha) [reply src/app/_components/SocialLink/SocialLink.js]: Done.

**Review by Dimirolz — APPROVED** [review]

File: src/app/_components/OutlinedBtn/OutlinedBtn.js
Comment (BigTako) [inline src/app/_components/OutlinedBtn/OutlinedBtn.js:1]: 🟠 - can we join SocialLink.js and InnerLink.js into single component?

File: src/app/_components/Header.js
Comment (BigTako) [inline src/app/_components/Header.js:1]: 🟠 - when we use component without children, we can use just `<Component />`, use <InnerLink ... /> and <SocialLink ... /> please

**Review by BigTako — CHANGES_REQUESTED** [review]

File: src/app/_components/OutlinedBtn/OutlinedBtn.js
Comment (BigTako) [reply src/app/_components/OutlinedBtn/OutlinedBtn.js:1]: Discussed in private network

File: src/app/_components/Header.js
Comment (BigTako) [reply src/app/_components/Header.js:1]: Discussed in private conversation.

**Review by BigTako — APPROVED** [review]

File: src/app/_components/OutlinedBtn/OutlinedBtn.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import p from 'prop-types';
+import cn from '@utils/cn';
+
+export default function OutlinedBtn({ children, className }) {
```
Comment (alex-popov-tech) [inline src/app/_components/OutlinedBtn/OutlinedBtn.js:5]: 🟠 isn't button should be part of separate PR with buttons?

File: src/app/_components/Footer.js [hunk 16]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import Link from 'next/link';
+import FooterImg from '@icons/white-logo.svg';
+import cn from '@utils/cn';
+import siteNav from '@config/siteNav';
+import OutlinedBtn from '@components/OutlinedBtn/OutlinedBtn';
+import SocialLink from '@components/Links/SocialLink';
+
+export default function Footer() {
+  const { links } = siteNav;
```
Comment (alex-popov-tech) [inline src/app/_components/Footer.js:10]: 🟠 
1. do you need those links somewhere in another place? if no, can these be placed somewhere near component?
2. please do not mutate data which you store somewhere, its not obvious behaviour for users which might want to update links list
